lib/ui/home_page/components/scanner_widget.dart

```diff
@@ -78,13 +78,22 @@ class _ScannerWidgetState extends State<ScannerWidget> {
                             child: ClipRRect(
                               borderRadius: BorderRadius.circular(30),
                               child: MobileScanner(
-                                allowDuplicates: false,
+                                // allowDuplicates: false,
                                 controller: MobileScannerController(
                                   facing: CameraFacing.back,
+                                  detectionSpeed: DetectionSpeed.normal,
                                   torchEnabled: false,
                                 ),
-                                onDetect: (barcode, args) {
-                                  if (barcode.rawValue == null) {
+                                onDetect: (capture) {
+                                  final List<Barcode> barcodes = capture.barcodes;
+                                  final barcode = barcodes.firstOrNull;
+
+                                  // TODO(NIK): What this should do is - for each QR code, determine whether or not
```

Comment:
Hum, I think the job of the scanner widget is to scan. Done. 

All the logic to check if we can handle it or not should be elsewhere